File: 4-medianTwoSortedArrays/Go/medianTwoSortedArrays.go
package medianTwoSortedArrays

import (
	"math"
	"sort"
)

func medianTwoSortedArrays(a1, a2 []int) float64 {
	a1 = append(a1, a2...)
	sort.Slice(a1, func(i, j int) bool {
		return a1[i] < a1[j]
	})

	mid := float64(len(a1)/2)

	if (isNotInt(mid)){
		return float64(a1[int(math.Floor(float64(mid)))])
	}
	return float64(a1[int(mid-1)] + a1[int(mid)])/2
}

func isNotInt(val float64) bool {
	return val == float64(int(val))
}

File: 4-medianTwoSortedArrays/Cpp/medianTwoSortedArrays.cpp
#include <vector>
#include <algorithm>
#include <cmath>

bool is_integer(float);

float medianTwoSortedArrays(std::vector<int> arr1, std::vector<int> arr2){
    auto size = arr1.size() + arr2.size();
    std::vector<int> arr_combined;
    arr_combined.reserve(size);
    arr_combined.insert(arr_combined.end(), arr1.begin(), arr1.end());
    arr_combined.insert(arr_combined.end(), arr2.begin(), arr2.end());
    std::sort( arr_combined.begin(), arr_combined.end());

    auto mid = size / 2.0;
    if(is_integer(mid)){
        return (arr_combined[mid-1] + arr_combined[mid])/2.0;
    }
    return arr_combined[mid];
}

bool is_integer(float k)
{
  return std::floor(k) == k;
}

